providers/go-feature-flag/src/main/java/org/gofeatureflag/provider/GoFeatureFlagProvider.java

```diff
@@ -0,0 +1,246 @@
+                }
+
+                boolean isPrimitive = expectedType == Boolean.class ||
+                        expectedType == String.class ||
+                        expectedType == Integer.class ||
+                        expectedType == Double.class;
+
+                // Convert the value received from the API.
+                T flagValue = isPrimitive
+                        ? goffResp.getValue() : (T) objectToValue(goffResp.getValue());
+
+                if (flagValue.getClass() != expectedType) {
+                    throw new TypeMismatchError("Flag value " + key + " had unexpected type " + flagValue.getClass() + ", expected " + expectedType + ".");
+                }
+
+                return ProviderEvaluation.<T>builder()
+                        .errorCode(goffResp.getErrorCode())
+                        .reason(mapReason(goffResp.getReason()))
+                        .value(flagValue)
+                        .variant(goffResp.getVariationType())
+                        .build();
+
+            }
+        } catch (IOException e) {
+            throw new GeneralError("unknown error while retrieving flag " + key);
+        }
+    }
+
+
+    /**
+     * mapReason is mapping the reason in string received by the API
+     * to our internal SDK reason enum.
+     *
+     * @param reason - string of the reason received from the API
+     * @return an item from the enum
+     */
+    private Reason mapReason(String reason) {
+        try {
+            return Reason.valueOf(reason);
+        } catch (IllegalArgumentException e) {
+            return Reason.UNKNOWN;
+        }
+    }
+
+
+    /**
+     * objectToValue is wrapping an object into a Value.
+     *
+     * @param object the object you want to wrap
+     * @return the wrapped object
+     */
+    private Value objectToValue(Object object) {
+        if (object instanceof Value) {
+            return (Value) object;
+        } else if (object instanceof String) {
+            return new Value((String) object);
+        } else if (object instanceof Boolean) {
+            return new Value((Boolean) object);
+        } else if (object instanceof Integer) {
+            return new Value((Integer) object);
+        } else if (object instanceof Double) {
+            return new Value((Double) object);
+        } else if (object instanceof Structure) {
+            return new Value((Structure) object);
+        } else if (object instanceof List) {
+            // need to translate each elem in list to a value
+            return new Value(((List<Object>) object).stream().map(this::objectToValue).collect(Collectors.toList()));
+        } else if (object instanceof Instant) {
+            return new Value((Instant) object);
+        } else if (object instanceof Map) {
+            return new Value(mapToStructure((Map<String, Object>) object));
+        } else {
+            throw new ClassCastException("Could not cast Object to Value");
```

Comment:
Do you want to handle `null` values? Since `null` is a valid JSON  and that is your content type, I think it should be allowed?

Reply:
GO Feature Flag is not sending `null` value, but I have added the test case to remove the key if we receive a null.